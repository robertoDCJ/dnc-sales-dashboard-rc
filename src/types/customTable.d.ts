/* eslint-disable no-undef */
export interface CustomTableProps {
  headers: string[]
  rows: React.ReactNode[][]
}
