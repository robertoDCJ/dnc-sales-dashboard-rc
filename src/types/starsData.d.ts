export interface StarsData {
  name: string
  subtitle: string
  value: number
}
