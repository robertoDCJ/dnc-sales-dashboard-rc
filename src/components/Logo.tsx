import { pxToRem } from '@/utils'
import styled from 'styled-components'

export const Logo = styled.figure<{ height: number; width: number }>`
  background-image: url(/${(props) => props.theme.appLogo});
  background-size: cover;
  height: ${(props) => pxToRem(props.height)};
  width: ${(props) => pxToRem(props.width)};
`
