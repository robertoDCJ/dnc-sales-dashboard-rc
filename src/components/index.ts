export { default as AvatarList } from './AvatarList'
export * from './BannerImage'
export * from './CardComponent'
export { default as CustomTable } from './CustomTable'
export { default as FormComponent } from './FormComponent'
export { default as Header } from './Header'
export * from './Logo'
export * from './StyledButton'
export * from './StyledInput'
export * from './Typographies'
