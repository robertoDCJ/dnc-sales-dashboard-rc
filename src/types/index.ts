export * from './appThemeContext'
export * from './avatarsList'
export * from './customChart'
export * from './customTable'
export * from './formComponent'
export * from './theme'
export * from './typographies'
