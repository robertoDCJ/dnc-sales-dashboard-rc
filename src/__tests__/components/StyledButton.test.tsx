/* eslint-disable no-undef */
import { StyledButton } from '@/components'
import { themesList } from '@/resources/themesList'
import { Theme } from '@/types'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

describe('StyledButton', () => {
  const renderComponent = (theme: Theme, className?: string, props = {}) =>
    render(
      <ThemeProvider theme={theme}>
        <StyledButton className={className} {...props} />
      </ThemeProvider>
    )

  themesList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
      it('should match the snapshot with alert class', () => {
        const { asFragment } = renderComponent(theme, 'alert')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match the snapshot with primary class', () => {
        const { asFragment } = renderComponent(theme, 'primary')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match the snapshot with borderless alert class', () => {
        const { asFragment } = renderComponent(theme, 'borderless-alert')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match the snapshot with disabled status', () => {
        const { asFragment } = renderComponent(theme, 'primary', {
          disabled: true,
        })
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
