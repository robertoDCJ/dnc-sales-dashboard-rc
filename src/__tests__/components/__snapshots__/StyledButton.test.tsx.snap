// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton Dark Theme should match the snapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6D7B8E;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6D7B8E;
}

<button
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Dark Theme should match the snapshot with borderless alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6D7B8E;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6D7B8E;
}

<button
    class="c0 borderless-alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Dark Theme should match the snapshot with disabled status 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6D7B8E;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6D7B8E;
}

<button
    class="c0 primary"
    disabled=""
  />
</DocumentFragment>
`;

exports[`StyledButton Dark Theme should match the snapshot with primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6D7B8E;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6D7B8E;
}

<button
    class="c0 primary"
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme should match the snapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #CCC;
  color: #666;
}

<button
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme should match the snapshot with borderless alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #CCC;
  color: #666;
}

<button
    class="c0 borderless-alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme should match the snapshot with disabled status 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #CCC;
  color: #666;
}

<button
    class="c0 primary"
    disabled=""
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme should match the snapshot with primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #CCC;
  color: #666;
}

<button
    class="c0 primary"
  />
</DocumentFragment>
`;
